### PR TITLE
[docs] update roadmap document

### PR DESCRIPTION
**Pull request by codeboten**
This document now contains the current focus of the maintainers of the collector project.

### CODE REVIEW

File: docs/ga-roadmap.md [hunk 1]
```diff
@@ -1 +1,54 @@
...
+* The Collector MUST be observable
+  * Metrics and traces SHOULD be produced for data in the hot path
+  * Metrics should be documented in the end-user documentation
+  * Metrics, or a subset of them, should be marked as stable in the documentation
+  * Logs SHOULD be produced in the stdout for Collector lifecycle events
+  * Telemetry should be stable, so that users can rely on that on for their dashboards and alerts
+* Move to make the collector better at dealing with backpressure and queueing
+  * Support backpressure from the exporter all the way back to the receiver
+  * Setting performance metrics in place and following best practices for benchmarking, load testing
+  * Enshrine in tests (generated for contrib) lifecycle expectations of components
```
Comment (songy23) [inline docs/ga-roadmap.md:43]: Maybe goleak tests as well?

File: docs/ga-roadmap.md [hunk 2]
```diff
@@ -1 +1,54 @@
...
+* End-user documentation should be provided as part of the official project’s documentation under opentelemetry.io, including:
```
Comment (songy23) [inline docs/ga-roadmap.md:44]: Do we want to call out queue, batch & backpressure too?

File: docs/ga-roadmap.md [hunk 3]
```diff
@@ -1 +1,54 @@
...
+- OTLP exporter
+- OTLP HTTP exporter
+
+These modules depend on a list of other modules, the full list is available in issue [#9375](https://github.com/open-telemetry/opentelemetry-collector/issues/9375).
+
+## Out of scope
+
+Explicitly, the following are not in the scope of v1 for the purposes of this document:
+
+* stabilization of all component developer APIs or other parts of the Collector. Component developers are NOT the audience for this work
```
Comment (TylerHelmuth) [inline docs/ga-roadmap.md:21]: Let's define what we mean by `component developer`. Are the core modules are required by component developers, and those are what we are stabilizing, so this exclusion isn't clear to me.

File: docs/ga-roadmap.md [hunk 4]
```diff
@@ -1 +1,54 @@
...
+* stabilization of additional components needed by distribution maintainers. Vendors are not the audience
+* Collector Builder
+* telemetrygen
+* Component APIs as part of the Core repository
```
Comment (TylerHelmuth) [inline docs/ga-roadmap.md:25]: Let's clarify how this is different from the first item.

File: docs/ga-roadmap.md [hunk 5]
```diff
@@ -1 +1,54 @@
...
+
+The proposed distribution is set to include the following components only:
+
+- OTLP receiver
+- OTLP exporter
+- OTLP HTTP exporter
+
+These modules depend on a list of other modules, the full list is available in issue [#9375](https://github.com/open-telemetry/opentelemetry-collector/issues/9375).
+
+## Out of scope
```
Comment (TylerHelmuth) [inline docs/ga-roadmap.md:17]: Does mdatagen need to be in this list?

File: docs/ga-roadmap.md (hunk 3 above)
Comment (codeboten) [reply docs/ga-roadmap.md]: You're right, i do think some of the APIs used by component developers are being stabilized as part of the work being done. Removing this line

File: docs/ga-roadmap.md (hunk 4 above)
Comment (codeboten) [reply docs/ga-roadmap.md]: tried to clarify by expanding the first item

File: docs/ga-roadmap.md (hunk 5 above)
Comment (codeboten) [reply docs/ga-roadmap.md:17]: added mdatagen

File: docs/ga-roadmap.md (hunk 2 above)
Comment (codeboten) [reply docs/ga-roadmap.md:43]: Added to the list of docs

**Review by TylerHelmuth — APPROVED** [review]

File: docs/ga-roadmap.md [hunk 6]
```diff
@@ -1 +1,54 @@
 # Collector v1 Roadmap
+
+This document contains the roadmap for the Collector as of 4-Apr-2024. The main goal of this roadmap is to provide clarity on the areas of focus in order to release a v1 of the Collector.
```
Comment (mx-psi) [inline docs/ga-roadmap.md:3]: nit: One can figure out the date by looking at the last commit that updated this file
```suggestion
This document contains the roadmap for the Collector. The main goal of this roadmap is to provide clarity on the areas of focus in order to release a v1 of the Collector.
```

File: docs/ga-roadmap.md [hunk 7]
```diff
@@ -1 +1,54 @@
...
+
+The proposed approach to delivering a stable release of the OpenTelemetry Collector is to produce a distribution of the Collector that contains a minimum set of components which have been stabilized. By doing so, the project contributors will ensure dependencies of those components have also been released under a stable version.
+
+The proposed distribution is set to include the following components only:
+
+- OTLP receiver
+- OTLP exporter
+- OTLP HTTP exporter
+
+These modules depend on a list of other modules, the full list is available in issue [#9375](https://github.com/open-telemetry/opentelemetry-collector/issues/9375).
```
Comment (mx-psi) [inline docs/ga-roadmap.md:15]:
```suggestion
These modules depend on a list of other modules, the full list is available in issue [#9375](https://github.com/open-telemetry/opentelemetry-collector/issues/9375).

All stabilized modules will conform to the API expectations outlined in the [VERSIONING.md](../VERSIONING.md) document.
```

File: docs/ga-roadmap.md [hunk 8]
```diff
@@ -1 +1,54 @@
...
+
+## Additional Requirements
+
+The following is a list of requirements for the Collector to be deemed as 1.0:
+
+* The Collector MUST be observable
+  * Metrics and traces SHOULD be produced for data in the hot path
+  * Metrics should be documented in the end-user documentation
+  * Metrics, or a subset of them, should be marked as stable in the documentation
+  * Logs SHOULD be produced in the stdout for Collector lifecycle events
```
Comment (mx-psi) [inline docs/ga-roadmap.md:37]: I believe we output to stderr by default. In any case, I don't think where we output the logs is important for this document, I would just say:
```suggestion
  * Logs SHOULD be produced for Collector lifecycle events
```

File: docs/ga-roadmap.md [hunk 9]
```diff
@@ -1 +1,54 @@
...
+  * Telemetry should be stable, so that users can rely on that on for their dashboards and alerts
```
Comment (mx-psi) [inline docs/ga-roadmap.md:38]: I think this is too restrictive and it contradicts the *"Metrics, or a subset of them, should be marked as stable in the documentation"* line. I would rephrase as:

```suggestion
  * Stability expectations and lifecycle for telemetry should be documented, so that users can know what they can rely on  for their dashboards and alerts
```

The important bit is that end-users understand what they can rely on, not that we stabilize every single metric from v1. Telemetry will evolve, and we will need something like [what Kubernetes does](https://kubernetes.io/blog/2021/04/23/kubernetes-release-1.21-metrics-stability-ga/#what-are-the-stability-levels-for-metrics)

File: docs/ga-roadmap.md [hunk 10]
```diff
@@ -1 +1,54 @@
...
+* Move to make the collector better at dealing with backpressure and queueing
```
Comment (mx-psi) [inline docs/ga-roadmap.md:39]: This is phrased differently from points above, we can argue what 'correctly' means but IMO we should phrase it closer to this
```suggestion
* The Collector should correctly handle backpressure and queueing
```

File: docs/ga-roadmap.md [hunk 11]
```diff
@@ -1 +1,54 @@
...
+  * Support backpressure from the exporter all the way back to the receiver
+  * Setting performance metrics in place and following best practices for benchmarking, load testing
+  * Enshrine in tests (generated for contrib) lifecycle expectations of components
```
Comment (mx-psi) [inline docs/ga-roadmap.md:42]: nit: The rest of the document starts all requirements with a noun and describes how it should behave. Can we rephrase these to be consistent? (e.g. "Backpressure should be supported [...]")

File: docs/ga-roadmap.md [hunk 12]
```diff
@@ -1 +1,54 @@
...
+* End-user documentation should be provided as part of the official project’s documentation under opentelemetry.io, including:
+  * Getting started with the Collector
+  * Available (stable) components and how to use them
+  * Blueprints for common use cases
+  * Error scenarios should be documented
+  * Specifically, error propagation
```
Comment (mx-psi) [inline docs/ga-roadmap.md:48]: These two points depend on each other and don't read like a title from a docs page as the rest do. I would rephrase as
```suggestion
  * Error scenarios and error propagation
```

File: docs/ga-roadmap.md [hunk 13]
```diff
@@ -1 +1,54 @@
...
+  * Troubleshooting, including how to obtain telemetry from the Collector for the purposes of bug reporting
```
Comment (mx-psi) [inline docs/ga-roadmap.md:49]: Same as above, rephrasing to make it more title-y
```suggestion
  * Troubleshooting and how to obtain telemetry from the Collector for the purposes of bug reporting
```

File: docs/ga-roadmap.md [hunk 14]
```diff
@@ -1 +1,54 @@
...
+  * Queueing, batching, and handling of backpressure
+* The Collector MUST be supported
+  * We need to make sure we can support the collector distribution and set clear expectations.
+  * We need to have workflows, processes in place to handle the influx of questions.
```
Comment (mx-psi) [inline docs/ga-roadmap.md:53]: I don't feel like the first line this is actionable, maybe something like this that merges the two points? (also rephrasing for consistency)
```suggestion
  * Processes, workflows and expectations regarding support, bug reporting and questions should be documented.
```

File: docs/ga-roadmap.md [hunk 15]
```diff
@@ -1 +1,54 @@
...
+  * We need to have a support policy in place for 1.0 with an end of life date.
```
Comment (mx-psi) [inline docs/ga-roadmap.md:54]: Are we going to have a date? I would say what we want, if anything, is a minimum timeline for support
```suggestion
  * A minimum support period for 1.0 is documented
```

but maybe just dropping this as a requirement is the way to go (I don't feel like we have a clear picture of what this means):

```suggestion
```

File: docs/ga-roadmap.md [hunk 16]
```diff
@@ -1 +1,54 @@
...
+* Collector Builder
+* telemetrygen
+* mdatagen
+* Operator
+
+Those components are free to pursue v1 at their own pace and may be the focus of future stability work.
+
+## Additional Requirements
+
+The following is a list of requirements for the Collector to be deemed as 1.0:
```
Comment (mx-psi) [inline docs/ga-roadmap.md:31]:
```suggestion
The following is a list of requirements for this minimal Collector distribution to be deemed as 1.0:
```

File: docs/ga-roadmap.md [hunk 17]
```diff
@@ -1 +1,54 @@
...
+
+* The Collector MUST be observable
```
Comment (mx-psi) [inline docs/ga-roadmap.md:33]: Should we just drop the uppercase MUST and SHOULD? We don't use them consistently (and I don't think that's because of normative vs non-normative) and this is not a specification document nor is it written in sufficient detail so as to make this actionable.

File: docs/ga-roadmap.md (hunk 15 above)
Comment (codeboten) [reply docs/ga-roadmap.md:55]: I wonder if this should be documented in the same way API/SDK stability is documented in the specification: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#api-support

File: docs/ga-roadmap.md (hunk 17 above)
Comment (codeboten) [reply docs/ga-roadmap.md]: Done

File: docs/ga-roadmap.md (hunk 15 above)
Comment (mx-psi) [reply docs/ga-roadmap.md:54]: That makes sense to me, we can link to that/paraphrase it here

**Review by mx-psi — APPROVED** [review]
